### PR TITLE
[#780] Cleanup nexus for CI

### LINKED ISSUE #780: Update nexus cleanup script with helm repo support
We should update nexus cleanup script so it can cleanup artifacts in nexus helm repository as well.

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This closes #780

### CODE REVIEW

**Review by botsman — DISMISSED** [review]

File: legion_test/bin/cleanup_nexus_repo [hunk 1]
```diff
@@ -37,59 +37,56 @@ Options = NamedTuple('Options', [
...
     try:
-        url = NEXUS_SEARCH_URL.format(host=nexus_url)
+        url = NEXUS_SEARCH_URL.format(host=options.nexus_url)
         log.debug('Fetching Nexus artifacts from URL {}'.format(url))
-        response = requests.get(url, params=search_filters, auth=(user, password))
-        if response.status_code not in allowed_responce_codes:
-                raise Exception("Non 200 response from Nexus!")
+        response = requests.get(url, params=search_filters, auth=(options.user, options.password))
+        if response.status_code not in ALLOWED_RESPONSE_CODES:
+                raise Exception('Non 200 response from Nexus!')
```
Comment (mcdoker18) [inline legion_test/bin/cleanup_nexus_repo]: Please, add `resp.status code` and `resp.text` to the exception message.

File: legion_test/bin/cleanup_nexus_repo [hunk 2]
```diff
@@ -37,59 +37,56 @@ Options = NamedTuple('Options', [
...
         while continuation_token is not None:
             log.debug('Fetching Nexus artifacts from URL {} (from {})'.format(url, continuation_token))
-            search_filters["continuationToken"] = continuation_token
-            response = requests.get(url, params=search_filters, auth=(user, password))
-            if response.status_code not in allowed_responce_codes:
-                raise Exception("Non 200 response from Nexus!")
+            search_filters['continuationToken'] = continuation_token
+            response = requests.get(url, params=search_filters, auth=(options.user, options.password))
+            if response.status_code not in ALLOWED_RESPONSE_CODES:
+                raise Exception('Non 200 response from Nexus!')
```
Comment (mcdoker18) [inline legion_test/bin/cleanup_nexus_repo]: the same

File: legion_test/bin/cleanup_nexus_repo [hunk 3]
```diff
@@ -102,119 +99,128 @@ def filter_builds_by_date(builds, remaining_days):
...
+                continue
+
+            regex_date = next(regex.group() for regex in build_date if regex)
+
+            if len(regex_date) == 12:
+                regex_pattern = '%y%m%d%H%M%S'
+            elif len(regex_date) == 14:
+                regex_pattern = '%Y%m%d%H%M%S'
+            else:
+                log.warning('Unknown length of matched date {!r}'.format(regex_date))
```
Comment (mcdoker18) [inline legion_test/bin/cleanup_nexus_repo:118]: If this branch executes than the regex_pattern variable will be undefined.

File: legion_test/bin/cleanup_nexus_repo [hunk 4]
```diff
@@ -102,119 +99,128 @@ def filter_builds_by_date(builds, remaining_days):
...
-                    host=nexus_url,
-                    resource=build.get("id")),
-                    auth=(user, password))
-                if response.status_code not in allowed_responce_codes:
-                    raise Exception("Non 200 response from Nexus!")
+                    host=options.nexus_url,
+                    resource=build.get('id')),
+                    auth=(options.user, options.password))
+                if response.status_code not in ALLOWED_RESPONSE_CODES:
+                    raise Exception('Non 200 response from Nexus!')
```
Comment (mcdoker18) [inline legion_test/bin/cleanup_nexus_repo]: same

**Review by mcdoker18 — CHANGES_REQUESTED** [review]

File: legion_test/bin/cleanup_nexus_repo (hunk 4 above)
Comment (kirillmakhonin) [reply legion_test/bin/cleanup_nexus_repo]: done

File: legion_test/bin/cleanup_nexus_repo (hunk 3 above)
Comment (kirillmakhonin) [reply legion_test/bin/cleanup_nexus_repo:118]: done

File: legion_test/bin/cleanup_nexus_repo (hunk 2 above)
Comment (kirillmakhonin) [reply legion_test/bin/cleanup_nexus_repo]: done

File: legion_test/bin/cleanup_nexus_repo (hunk 1 above)
Comment (kirillmakhonin) [reply legion_test/bin/cleanup_nexus_repo]: done

**Review by mcdoker18 — APPROVED** [review]

**Review by botsman — APPROVED** [review]

**Review by alexander-semenets — APPROVED** [review]

**Pull request merged**